p2p/pex/pex_reactor.go

```diff
@@ -241,7 +247,10 @@ func (r *PEXReactor) ReceiveAddrs(addrs []*p2p.NetAddress, src Peer) error {
 
 	srcAddr := src.NodeInfo().NetAddress()
 	for _, netAddr := range addrs {
-		if netAddr != nil {
+		if netAddr == nil {
+			continue
+		}
+		if !isAddrPrivate(netAddr, r.config.PrivatePeerIDs) {
```

Comment:
Minor nit, as the loop doesn't have much in it except for the if clause and `r.book.AddAddress(netAddr, srcAddr)`
```go
if netAddr != nil && !isAddrPrivate(netAddr, r.config.PrivatePeerIDs) {
```